ValveKeyValue/ValveKeyValue.Test/Helpers/BlockingStream.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.IO;
+using System.Text;
+
+namespace ValveKeyValue.Test.Helpers;
+
+/// <summary>
+///     Example implementation of <see cref="Stream" /> that reads at most <see cref="MaxReadAtOnce" /> characters during a
+///     single <see cref="Read" /> operation, blocking in the process.
+/// </summary>
+internal sealed class BlockingStream : Stream
+{
+    private static readonly byte[] TestData = Encoding.UTF8.GetBytes(
```

Comment:
IMO this would be better as a wrapper for an existing `Stream` than as a weird stream with hardcoded content.